src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py

```diff
@@ -274,6 +278,36 @@ def getDatasetLocations(self, datasets, account=None):
                                                              account=account)
             result[datasetPath] = self.cric.PNNstoPSNs(locations)
         return result
+    
+    def getDatasetLocationsFromMSPileup(self, datasetsWithDbsURL):
+        """
+        Returns a dictionary with the location of the datasets according to MSPileup
+        The definition of "location" here is a union of all sites holding at least
+        part of the dataset (defined by the DATASET grouping).
+        :param datasets: list of datasets
+        :param account: rucio account to use, if it is not provided we fallback to
+        pileup account for backward compatibility
+        :return: a dictionary of dataset locations, key'ed by the dataset name
+        """
+        if isinstance(datasetsWithDbsURL, str):
+            datasetsWithDbsURL = [datasetsWithDbsURL]
+        result = {}
+        for dataset, dbsUrl in zip(datasetsWithDbsURL):
+            try:
+                queryDict = {'pileupName': dataset,
+                             'filters': ['expectedRSEs', 'currentRSEs', 'pileupName', 'containerFraction', 'ruleIds']}
+                instance = '-testbed' if 'testbed' in dbsUrl else '-prod'
+                msPileupUrl = f"https://cmsweb{instance}.cern.ch/ms-pileup/data/pileup"
+                doc = getPileupDocs(msPileupUrl, queryDict)
```

Comment:
The output of this call is likely something like:
```
{"result": [{pileupA data}, {pileupB data}, ...]}

Reply:
That means you need to iterate through the output of that API and compare it against the pileups that we are updating the location, provided in `datasetsWithDbsURL`